### PR TITLE
ci: 1770 update ci workflows for best practices and standards round2

### PR DESCRIPTION
**Pull request by rbarkerSL**
**Description**:

This PR modifies the workflows  in order to support best practices and standards on the repo.

**Related issue(s)**: #1770 

Fixes #

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -1,2 +1,40 @@
 # Default code owners for entire repository
-*   @agadzhalov @thenswan @SimiHunjan
+*   @agadzhalov @thenswan @SimiHunjan @hashgraph/release-engineering @hashgraph/release-engineering-managers
```
Comment (thenswan) [inline .github/CODEOWNERS:2]: We need to add `0xivanov` as he is also repo maintainer.

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -1,2 +1,40 @@
...
+
+#########################
+#####  Core Files  ######
+#########################
+
+# NOTE: Must be placed last to ensure enforcement over all other rules
```
Comment (thenswan) [inline .github/CODEOWNERS:8]: Is it mandatory to restrict those files from above repo maintainers?

**Review by thenswan — COMMENTED** [review]
Would like to understand couple points, thanks 🙏🏻

File: .github/CODEOWNERS (hunk 2 above)
Comment (thenswan) [reply .github/CODEOWNERS:8]: cc @nathanklick

File: .github/CODEOWNERS (hunk 1 above)
Comment (thenswan) [reply .github/CODEOWNERS:2]: cc @nathanklick

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -1,2 +1,40 @@
...
+
+# Protection Rules for Github Configuration Files and Actions Workflows
+/.github/                                       @hashgraph/release-engineering @hashgraph/release-engineering-managers @0xivanov @MrValioBg
```
Comment (agadzhalov) [inline .github/CODEOWNERS:11]: why would we have explicitly **0xivanov** and **MrValioBg** here? I don't see any reason for that. Also **MrValioBg** is not part from our SDK team so he shouldn't be included

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -136,8 +151,8 @@ jobs:
 
       - name: Prepare .env for Examples
         run: |
-          echo "OPERATOR_KEY=0xa608e2130a0a3cb34f86e757303c862bee353d9ab77ba4387ec084f881d420d4" > examples/.env
-          echo "OPERATOR_ID=0.0.1022" >> examples/.env
+          echo "OPERATOR_KEY=${{ secrets.OPERATOR_KEY }}" > examples/.env
```
Comment (thenswan) [inline .github/workflows/build.yml:154]: Seems it is not working, because tests and examples are failing now with `PAYER_ACCOUNT_NOT_FOUND`

Comment (rbarkerSL) [reply .github/workflows/build.yml]: reverted

**Review by thenswan — CHANGES_REQUESTED** [review]
Couple things:
- `/.github/` folder is still restricted, and I regularly update the build file in it to run SDK tests and examples against different service tags;
- maybe it would be good to follow [C++ SDK example](https://github.com/hashgraph/hedera-sdk-cpp/blob/main/.github/CODEOWNERS) and create a group `hashgraph/hedera-sdk-java-maintainers` for `agadzhalov thenswan SimiHunjan 0xivanov`.

**Review by thenswan — APPROVED** [review]
LGTM, thanks @rbarkerSL

**Pull request merged**